internal/api/utils.go

```diff
@@ -752,3 +757,32 @@ func GenerateAndStreamCSV(w io.Writer, recommendationSets []model.Recommendation
 	}
 	return nil
 }
+
+func resolveResponseFormat(acceptHeaderVal string, formatQueryParamVal string) (string, error) {
+	if acceptHeaderVal == "" && formatQueryParamVal == "" {
+		return "json", nil // default format
+	}
+
+	responseFormat := ""
+	switch acceptHeaderVal {
+	case "text/csv":
+		responseFormat = "csv"
+	case "application/json":
+		responseFormat = "json"
+	}
+
+	if responseFormat != "" {
+		return responseFormat, nil // preferring header value
+	} else {
```

Comment:
What is the expected behavior here when someone passes accept header's value other than `text/csv` or `application/json` and `format` as `json` or `csv` ?  

@Aurobinda55  do you think this scenario should also be captured in a test?

Reply:
The `accept/Accept` header value will checked first if value is invalid/empty, the format param value will be evaluated.
Let's say request has `"Accept": "application/json"` + format=csv, response will be provided in JSON.
Documented, [here](https://github.com/RedHatInsights/ros-ocp-backend/pull/419/files#diff-a9865368a7fc7fa33065e35b2343f10d08fb79d65205435403d0a163a3044713R29).